src/visualization/ColorPalette/ColorMap.cc

```diff
@@ -0,0 +1,284 @@
+    int numbOfKmer = 0;
+    for (int i=0;i<SequenceSizes.size();i++)
+        numbOfKmer += SequenceSizes[i];
+    updateColorList(numbOfKmer);
+    ///< This variable is the output List
+    vector<string> kmerDeclarations;
+
+    ///< a boolean for the "got this kmer a color?"-Check
+    bool matched = false;
+    ///< a index for the k-mers per sequence
+    int index;
+    
+    /**
+     * This part of the method will push all Kmers in the List, which are not in the list already.
+     */
+    for (int i=0;i<Sequences.size();i++) {
+        index = 0;
+        while (index<SequenceSizes[i]) {
+            ///< Comparing all k-mers to know if the Kmer is already in the list
+            for (int j=0;j<kmerDeclarations.size();j++) {
+                if (kmerDeclarations[j]==Sequences[i].substr(index*k, k)) {
+                    matched = true;
+                    break;
+                }
+            }
+            ///< If its a new kmer, so we push it here
+            if (!matched)
+                kmerDeclarations.push_back(Sequences[i].substr(index*k, k));
+            matched = false;
+            index += 1;
+        }
+    }
+    differentKmers = kmerDeclarations;
+}
+
+/**
+ * This Method will make a vector of Colors in terms of a number, which defines the Length of the 
+ * length of the List. Then it will update our ColorList with it. In our case the argument 
+ * describes the number of Kmers we are looking for to map.
+ */
+void ColorMap::updateColorList(int numbOfKmer){
+    vector<Color> returnList; ///< the list i use to adjust this as the new List
+    Color placeholder; ///< A placeholder which is needed to programm with rgb values
+    /**
+     * In this part of the method i push Colors in the List which i searched in the
+     * web, so that they are well different.
+     */
+    placeholder.r = 255;
+    placeholder.g = 0;
+    placeholder.b = 0;
+    returnList.push_back(placeholder);
+    placeholder.g = 128;
+    returnList.push_back(placeholder);
+    placeholder.g = 255;
+    returnList.push_back(placeholder);
+    placeholder.r = 128;
+    returnList.push_back(placeholder);
+    placeholder.r = 0;
+    returnList.push_back(placeholder);
+    placeholder.b = 128;
+    returnList.push_back(placeholder);
+    placeholder.b = 255;
+    returnList.push_back(placeholder);
+    placeholder.g = 128;
+    returnList.push_back(placeholder);
+    placeholder.g = 0;
+    returnList.push_back(placeholder);
+    placeholder.r = 128;
+    returnList.push_back(placeholder);
+    placeholder.r = 255;
+    returnList.push_back(placeholder);
+    placeholder.b = 128;
+    returnList.push_back(placeholder);
+    placeholder.r = 128;
+    placeholder.g = 128;
+    placeholder.b = 128;
+    returnList.push_back(placeholder);
+    /**
+     * In this part of the method i push Colors in terms of a algorithm i made for myself.
+     * This algorithm is maybe not completed yet.
+     */
```

Comment:
There could be very similar colors to the hard-coded colors above, couldn't there?
I suggest you do a crude color-similarity check, and do not use two very similar colors.